### ISSUE TITLE
cuda runtime error SplineConv

### ISSUE THREAD

**Issue by milts10** [rusty1s/pytorch_geometric#615]
## 🐛 Bug
Hello, thank you very much for this tool is amazing! 

I have one main problem which is cuda runtime error, if I use a model with only two spline convolutions 
`
class Net(torch.nn.Module):

    def __init__(self):

        super(Net, self).__init__()

        self.conv1 = SplineConv(d.num_features, 32, dim=2, kernel_size=5)

        self.conv2 = SplineConv(32, 2, dim=2, kernel_size=5)

`
Everything works and I never get an error 

If on the other hand, use a model with 3 convolutions

`

class Net(torch.nn.Module):

    def __init__(self):

        super(Net, self).__init__()

        self.conv1 = SplineConv(d.num_features, 32, dim=2, kernel_size=5)

        self.conv2 = SplineConv(32, 16, dim=2, kernel_size=5)

        self.conv3 = SplineConv(16, 2, dim=2, kernel_size=5)

`
Then I get cuda runtime error, I try to run  

CUDA_LAUNCH_BLOCKING=1 python mnist_voxel_grid.py

as you suggested on another question and this is the error that I am getting 

"cuda runtime error (77) : an illegal memory access was encountered at /opt/conda/conda-bld/pytorch_1556653114079/work/aten/src/THC/generic/THCTensorMath.cu:26"

Do you have any ideas how to solve this I am really stack I tried everything for a week now

Thank you for your time

## Environment

* OS: linux
* Python version:  Python 3.7.3
* PyTorch version: 1.1.0
* CUDA/cuDNN version: 9.0

**Comment by rusty1s** [comment]
Do you call the operators in the right order? You can also try to enable debug mode with `torch_geometric.debug()` at the too of your script.

**Comment by milts10** [comment]
wow first of all really fast response! 

Yea I am sure that I call them with the right order, I tried to just copy and past torch_geometric.debug() and doesn't do something but to be honest I don't know what I am waiting to see

Oh something else that can possibly help is that I am using the quare distance error 
`
temp = result - data.y

temp = temp + 0.001

loss = torch.norm(temp,p=2,dim=1).mean()

print(loss)
`

**Comment by rusty1s** [comment]
The debug call evaluates the correctness of your data and checks that the SplineConv arguments match with the initialized parameters like in_channels. I do not think it is due to the loss function because it works with two layers.

**Comment by milts10** [comment]
Ok, I just found the problem. 

Because I was creating my dataset I had edge_attr from [-1,1] which make sense mathematically but is wrong based on your paper which says that edge_attr is from [0,1] and I just notice that now.

So for some reason with 2 layers, the system was kind of working but with 3+ I always had NaN values or CUDA runtime error, so it was really difficult for me to find the error.


Now that we are discussing that before I close the issue.
Let's say for the MNIST superpixels when it comes to the coordinates of the edges. You just calculate the difference between the cartesian coordinates of the center superpixels which can be a range of [-20,20] and then you translate everything to [0,1] or you are doing something special?

Thank you

**Comment by rusty1s** [comment]
That is exactly what we are doing with the `torch_geometric.transforms.Cartesian` transform.

**Comment by milts10** [comment]
Great, I didn't know about that.

Again thank you very much for everything, amazing paper amazing tool amazing response!!! 

Have a great day
